src/dir.rs

```diff
@@ -0,0 +1,210 @@
+use {Error, NixPath, Result};
+use errno::Errno;
+use fcntl::{self, OFlag};
+use libc;
+use std::os::unix::io::{AsRawFd, IntoRawFd, RawFd};
+use std::{ffi, fmt, ptr};
+use sys;
+
+#[cfg(target_os = "linux")]
```

Comment:
Can we use the `cfg_if!` macro here instead? It's a little more verbose but I find it easier to read an if-else rather than trying to determine that through consecutive `cfg` statements.

Reply:
likewise